src/main/kotlin/com/openlattice/organizations/ExternalDatabaseManagementService.kt

```diff
@@ -271,12 +368,30 @@ class ExternalDatabaseManagementService(
         updateHBARecords(dbName)
     }
 
+    fun getOrganizationOwners(orgId: UUID): Set<SecurablePrincipal> {
+        val principals = securePrincipalsManager.getAuthorizedPrincipalsOnSecurableObject(AclKey(orgId), EnumSet.of(Permission.OWNER))
+        return securePrincipalsManager.getSecurablePrincipals(principals).toSet()
+    }
+
     /**
      * Sets privileges for a user on an organization's table or column
      */
     fun executePrivilegesUpdate(action: Action, acls: List<Acl>) {
-        val aclsByOrg = acls.groupBy { it.aclKey[0] }
-        aclsByOrg.forEach { (orgId, orgAcls) ->
+        //split acls between table and column objects
+        val aclsByType = acls.partition { it.aclKey.size == 1 }
+
+        val tableAclsByOrg = aclsByType.first.groupBy {
+            organizationExternalDatabaseTables.getValue(it.aclKey[0]).organizationId
+        }
+        val columnAclsByOrg = aclsByType.second.groupBy {
+            organizationExternalDatabaseColumns.getValue(it.aclKey[1]).organizationId
+        }
+        val orgIds = tableAclsByOrg.keys.union(columnAclsByOrg.keys)
```

Comment:
You could just do: `val orgIds = tableAclsByOrg.keys + columnAclsByOrg.keys`

Reply:
is that better?

Reply:
Well, `union` will first create a `LinkedHashSet` with the elements of the first collection (`tableAclsByOrg.keys`), so it will have an expected size of that and then add all the elements from the second collection (`columnAclsByOrg.keys`), whereas `plus` will create an empty `LinkedHashSet` with the expected size of both collections and then add all elements from both collections. So if there are more elements in the second collection, it is better to use `plus`

```
public infix fun <T> Iterable<T>.union(other: Iterable<T>): Set<T> {
    val set = this.toMutableSet()
    set.addAll(other)
    return set
}
```

```
public operator fun <T> Set<T>.plus(elements: Iterable<T>): Set<T> {
    val result = LinkedHashSet<T>(mapCapacity(elements.collectionSizeOrNull()?.let { this.size + it } ?: this.size * 2))
    result.addAll(this)
    result.addAll(elements)
    return result
}
```

Reply:
Hmm well we won't know which collection has more elements so I can switch to plus to be safe